tools/clang/test/DXILValidation/atomics.hlsl

```diff
@@ -0,0 +1,53 @@
+// Validation template including valid operations and invalid destinations
+// that the validator test will swap in to ensure that the validator produces
+// appropriate errors.
+
+// Valid resources for atomics to create valid output that will later be manipulated to test the validator
+RWStructuredBuffer<uint> rw_structbuf;
+RWBuffer<uint> rw_buf;
+RWTexture1D<uint> rw_tex;
+
+// SRVs to plug into atomics
+StructuredBuffer<uint> ro_structbuf;
+Buffer<uint> ro_buf;
+Texture1D<uint> ro_tex;
+
+const groupshared uint cgs_var;
```

Comment:
What does this produce?  Should we even allow const groupshared global?

Reply:
It produces as valid code as can be involving an uninitialized variable. Should we? We certainly have in the past. Fxc accepts it if it's initialized. Of course, that produces an error in dxc because groupshared inits aren't supported.

Reply:
In FXC, while it accepts it, does it actually produce a groupshared declaration in the bytecode for it?  That was my concern.  const groupshared could always just be turned into const global data which could either become immediate values or an immediate constant buffer if dynamically indexed.  Hence the groupshared would basically get ignored, so if you declared it this way, it might be useful to at least warn that the groupshared will basically be ignored.

For uninitialized constant (internal-linkage global or local) variable, we should probably fail on declaration or use of such, since it's impossible for it to be anything but undef.

Reply:
I have an issue on me to correctly initialize groupshared, which would make this code valid. I'm not sure what we should do, but I think it's outside the scope of this change.